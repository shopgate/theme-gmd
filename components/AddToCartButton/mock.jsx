/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
/**
 * Mocked AddToCartButton.
 * @type {MockedAddToCartButton}
 */
// eslint-disable-next-line react/prefer-stateless-function
export const MockedAddToCartButton = class extends Component {
  static propTypes = {
    handleAddToCart: PropTypes.func,
  };
  static defaultProps = {
    handleAddToCart: () => {},
  };
  /**
   * Renders mocked button.
   * @return {JSX}
   */
  render() {
    return (
      <button onClick={this.props.handleAddToCart} />
    );
  }
};
