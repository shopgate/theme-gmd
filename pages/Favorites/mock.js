/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */
const foo = {
  productData: {
    name: 'Foo',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'foo',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

const bar = {
  productData: {
    name: 'Bar',
    availability: {
      text: 'Available',
      state: 'OK',
    },
    id: 'bar',
    baseProductId: null,
    price: {
      unitPriceStriked: 100,
      unitPriceWithTax: 101,
      currency: 'EUR',
    },
  },
};

export const mockedEmptyState = {
  product: {
    productsById: {
      foo,
    },
  },
  favorites: {
    products: {
      ids: [],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedState = {
  ...mockedEmptyState,
  favorites: {
    products: {
      ids: ['foo'],
      ready: true,
      isFetching: false,
    },
  },
};

export const mockedNotReadyState = {
  ...mockedEmptyState,
  favorites: {
    products: {},
    isFetching: false,
  },
};

export const mockedNextProps = {
  products: [
    foo.productData,
    bar.productData,
  ],
};

