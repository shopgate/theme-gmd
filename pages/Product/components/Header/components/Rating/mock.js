/*
 *  Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 *  This source code is licensed under the Apache 2.0 license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */

/**
 * Mock for getElementById
 * @param {function} scrollSpy Element.scroll spy function.
 * @returns {function}
 */
const getElementById = scrollSpy => () => ({
  offsetTop: 100,
  closest() {
    return {
      scroll: scrollSpy,
    };
  },
});

export {
  getElementById,
};
