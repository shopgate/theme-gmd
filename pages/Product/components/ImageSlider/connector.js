/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { connect } from 'react-redux';
import {
  getProductImages,
  getCurrentBaseProduct,
} from '@shopgate/pwa-common-commerce/product/selectors/product';
import openGallery from '../../actions/openGallery';

/**
 * Maps the contents of the state to the component props.
 * @param {Object} state The current application state.
 * @return {Object} The extended component props.
 */
const mapStateToProps = state => ({
  images: getProductImages(state),
  product: getCurrentBaseProduct(state),
});

/**
 * Connects the dispatch function to a callable function in the props.
 * @param {Function} dispatch The redux dispatch function.
 * @return {Object} The extended component props.
 */
const mapDispatchToProps = dispatch => ({
  onOpen: (productId, currentSlide) => dispatch(openGallery(productId, currentSlide)),
});

export default connect(mapStateToProps, mapDispatchToProps);
