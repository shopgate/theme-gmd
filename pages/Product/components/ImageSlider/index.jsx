import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Hammer from '@shopgate/react-hammerjs';
import ProductImage from 'Components/ProductImage';
import BaseImageSlider from '@shopgate/pwa-ui-shared/ImageSlider';
import connect from './connector';

const resolutions = [
  {
    width: 440,
    height: 440,
  },
  {
    width: 1024,
    height: 1024,
  },
];

/**
 * The product image slider component.
 * @param {number} currentSlide The index of the current visible slide.
 * @param {Object} props The component props.
 * @param {Object} props.product Basic product data from the product state.
 * @param {Array} props.images Array of image urls.
 */
class ImageSlider extends PureComponent {
  static propTypes = {
    images: PropTypes.arrayOf(PropTypes.string),
    navigate: PropTypes.func,
    product: PropTypes.shape(),
  };

  static defaultProps = {
    images: null,
    product: null,
    navigate: () => {},
  };

  currentSlide = 0;

  handleOpenGallery = () => {
    const { images } = this.props;

    if (!images || !images.length) {
      return;
    }

    this.props.navigate(this.currentSlide);
  };

  handleSlideChange = (currentSlide) => {
    this.currentSlide = currentSlide;
  };

  /**
   * Renders the product image slider component.
   * @returns {JSX}
   */
  render() {
    const { product, images } = this.props;

    let content = null;

    if (!product || !images || images.length === 0) {
      content = (
        <ProductImage
          src={product ? product.featuredImageUrl : null}
          forcePlaceholder={!product}
          resolutions={resolutions}
        />
      );
    } else {
      content = (
        <BaseImageSlider loop indicators onSlideChange={this.handleSlideChange}>
          {images.map(image => (
            <ProductImage key={image} src={image} animating={false} resolutions={resolutions} />
          ))}
        </BaseImageSlider>
      );
    }

    return (
      <Hammer
        onPinchStart={this.handleOpenGallery}
        onTap={this.handleOpenGallery}
        direction="DIRECTION_ALL"
        options={{
          touchAction: 'pan-x pan-y',
          recognizers: {
            pinch: { enable: true },
          },
        }}
      >
        <div>{content}</div>
      </Hammer>
    );
  }
}

export default connect(ImageSlider);
