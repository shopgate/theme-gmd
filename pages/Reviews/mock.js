/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {
  mockedStateWithoutReview,
} from 'Components/Reviews/mock';

export const mockedState = {
  ...mockedStateWithoutReview,
  reviews: {
    reviewsByHash: {},
  },
  history: {
    state: {
      title: '',
      viewTop: true,
    },
  },
};
