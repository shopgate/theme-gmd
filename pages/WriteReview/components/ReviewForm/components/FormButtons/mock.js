/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Mocked state with product only. Reviews not fetched.
 * @type {Object}
 */
export const mockedState = {
  history: {
    pathname: '/item/666f6f/write_review',
  },
  view: {
    isLoading: false,
  },
};
