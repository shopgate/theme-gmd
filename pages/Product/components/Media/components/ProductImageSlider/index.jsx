import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import isEqual from 'lodash/isEqual';
import noop from 'lodash/noop';
import { withNavigation, bin2hex } from '@shopgate/engage/core';
import { Swiper, Portal } from '@shopgate/engage/components';
import {
  PRODUCT_IMAGE,
  PRODUCT_IMAGE_AFTER,
  PRODUCT_IMAGE_BEFORE,
  loadProductImage,
  ITEM_PATH,
  ProductImage,
  getProductImageSettings,
} from '@shopgate/engage/product';
import MediaSection from '../MediaSection';
import connect from './connector';

/**
 * The product image slider component.
 * @param {number} currentSlide The index of the current visible slide.
 * @deprecated since catalog 2.0
 */
class ProductImageSlider extends Component {
  static propTypes = {
    'aria-hidden': PropTypes.bool,
    className: PropTypes.string,
    historyPush: PropTypes.func,
    images: PropTypes.arrayOf(PropTypes.string),
    product: PropTypes.shape(),
  };

  static defaultProps = {
    'aria-hidden': null,
    className: null,
    historyPush: noop,
    images: null,
    product: null,
  };

  /**
   * @param {Object} nextProps the next props
   * @returns {boolean}
   */
  shouldComponentUpdate(nextProps) {
    let depImage = null;
    if (this.props.product !== nextProps.product) {
      // Update only after new product data is received, skipping null
      if (nextProps.product && nextProps.product.featuredImageBaseUrl) {
        depImage = nextProps.product.featuredImageBaseUrl;
      }
    }

    if (this.props.images && nextProps.images) {
      if (isEqual(this.props.images, nextProps.images)) {
        return false;
      }

      if (this.props.images.length !== nextProps.images.length && !nextProps.images.length) {
        // Product image will be shown due to no images
        return true;
      }

      [depImage] = nextProps.images;
    }

    if (depImage) {
      // We have dependency image that we need to load before re-render
      loadProductImage(depImage).then(() => this.forceUpdate());
    }

    return false;
  }

  handleOpenGallery = () => {
    this.props.historyPush({
      pathname: `${ITEM_PATH}/${bin2hex(this.props.product.id)}/gallery/${this.currentSlide}`,
    });
  };

  handleSlideChange = (currentSlide) => {
    this.currentSlide = currentSlide;
  };

  currentSlide = 0;

  /**
   * Renders the product image slider component.
   * @returns {JSX}
   */
  render() {
    const {
      product, images, 'aria-hidden': ariaHidden, className,
    } = this.props;
    const { HeroImage: pdpResolutions } = getProductImageSettings();
    let content;
    let onClick = this.handleOpenGallery;

    if (product && Array.isArray(images) && images.length > 1) {
      content = (
        <Swiper
          loop
          indicators
          onSlideChange={this.handleSlideChange}
          className={className}
        >
          {images.map(image => (
            <Swiper.Item key={`${product.id}-${image}`}>
              <ProductImage
                src={image}
                animating={false}
                resolutions={pdpResolutions}
                noBackground
              />
            </Swiper.Item>
          ))}
        </Swiper>
      );
    }

    if (!content) {
      content = (
        <ProductImage
          src={product ? product.featuredImageBaseUrl : null}
          className={className}
          forcePlaceholder={!product}
          resolutions={pdpResolutions}
        />
      );
      if (!images || (images && !images.length)) {
        onClick = noop;
      }
    }

    return (
      <Fragment>
        <Portal name={PRODUCT_IMAGE_BEFORE} />
        <Portal name={PRODUCT_IMAGE}>
          <MediaSection
            product={product}
            aria-hidden={ariaHidden}
            onClick={onClick}
          >
            {content}
          </MediaSection>
        </Portal>
        <Portal name={PRODUCT_IMAGE_AFTER} />
      </Fragment>
    );
  }
}

export default withNavigation(connect(ProductImageSlider));
