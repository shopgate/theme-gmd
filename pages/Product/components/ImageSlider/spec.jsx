import React from 'react';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import ProductImage from 'Components/ProductImage';
import ImageSlider from '@shopgate/pwa-ui-shared/ImageSlider';
import { initialState } from '@shopgate/pwa-common-commerce/product/mock';
import { basicProductState } from '../mock';
import ProductImageSlider from './index';

jest.mock('@shopgate/react-hammerjs/src/Hammer', () => ({ children }) => children);

describe('<ProductImageSlider />', () => {
  const mockStore = configureStore([thunk]);
  const product = {
    featuredImageUrl: 'featured.png',
  };

  it('should render image placeholder if no data is available', () => {
    const store = mockStore(initialState);
    const wrapper = mount(<ProductImageSlider store={store} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('img').length).toEqual(0);
    expect(wrapper.find(ProductImage).length).toEqual(1);
    expect(wrapper.find(ImageSlider).length).toEqual(0);
  });

  it('should render featured image if only product data a is available', () => {
    const { productsById } = basicProductState.product;
    const { productId } = basicProductState.product.currentProduct;
    productsById[productId].productData.featuredImageUrl = product.featuredImageUrl;

    const state = {
      ...basicProductState,
      product: {
        ...basicProductState.product,
        productsById,
        imagesByProductId: {},
      },
    };
    const store = mockStore(state);
    const wrapper = mount(<ProductImageSlider store={store} productId={productId} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(ProductImage).length).toEqual(1);
    expect(wrapper.find('img').length).toEqual(0);
    expect(wrapper.find(ProductImage).at(0).prop('src')).toEqual('featured.png');
    expect(wrapper.find(ImageSlider).length).toEqual(0);
  });

  it('should render image slider if all data is available', () => {
    const { productId } = basicProductState.product.currentProduct;

    const store = mockStore(basicProductState);
    const wrapper = mount(<ProductImageSlider store={store} productId={productId} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(ImageSlider).length).toEqual(1);

    basicProductState.product.imagesByProductId[productId].images.forEach((url, index) => {
      expect(wrapper.find(ImageSlider).find(ProductImage).at(index).prop('src')).toEqual(url);
    });
  });

  it('should open gallery', () => {
    const { productId } = basicProductState.product.currentProduct;

    const store = mockStore(basicProductState);
    const wrapper = mount(<ProductImageSlider store={store} productId={productId} />);
    wrapper.find('ImageSlider').at(0).instance().handleOpenGallery();
    const navigateAction = [{
      type: 'NAVIGATE',
      action: 'PUSH',
      location: '/item/393133/gallery/0',
      state: undefined,
    }];

    expect(store.getActions()).toEqual(navigateAction);
  });
});
