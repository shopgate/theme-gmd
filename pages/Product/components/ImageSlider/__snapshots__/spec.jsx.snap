// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductImageSlider /> should render featured image if only product data a is available 1`] = `
<Connect(ImageSlider)
  productId="913"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ImageSlider
    images={null}
    navigate={[Function]}
    product={
      Object {
        "ageRating": 0,
        "availability": Object {
          "state": "ok",
          "text": "In stock",
        },
        "baseProductId": null,
        "featuredImageUrl": "featured.png",
        "flags": Object {
          "hasChildren": false,
          "hasOptions": false,
          "hasVariants": false,
        },
        "highlight": true,
        "id": "913",
        "identifiers": Object {
          "sku": "857347363",
        },
        "liveshoppings": Array [],
        "manufacturer": "",
        "name": "This is a product with a very very long name -5-",
        "price": Object {
          "currency": "EUR",
          "info": "",
          "msrp": 0,
          "taxAmount": 0,
          "taxPercent": 0,
          "tiers": Array [],
          "unitPrice": 89,
          "unitPriceMax": 1000,
          "unitPriceMin": 0,
          "unitPriceNet": 89,
          "unitPriceStriked": 0,
          "unitPriceWithTax": 89,
        },
        "rating": Object {
          "average": 0,
          "count": 0,
          "reviewCount": 0,
        },
        "stock": Object {
          "ignoreQuantity": true,
          "info": "In stock",
          "maxOrderQuantity": 10000,
          "minOrderQuantity": 1,
          "orderable": true,
          "quantity": 0,
        },
      }
    }
    productId="913"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <Component
      direction="DIRECTION_ALL"
      onPinchStart={[Function]}
      onTap={[Function]}
      options={
        Object {
          "recognizers": Object {
            "pinch": Object {
              "enable": true,
            },
          },
          "touchAction": "pan-x pan-y",
        }
      }
    >
      <div>
        <ProductImage
          alt={null}
          animating={true}
          forcePlaceholder={false}
          highestResolutionLoaded={[Function]}
          ratio={null}
          resolutions={
            Array [
              Object {
                "height": 440,
                "width": 440,
              },
              Object {
                "height": 1024,
                "width": 1024,
              },
            ]
          }
          src="featured.png"
        >
          <Image
            alt={null}
            animating={true}
            backgroundColor="#fff"
            className=""
            forcePlaceholder={false}
            highestResolutionLoaded={[Function]}
            onError={[Function]}
            onLoad={null}
            ratio={null}
            resolutions={
              Array [
                Object {
                  "height": 440,
                  "width": 440,
                },
                Object {
                  "height": 1024,
                  "width": 1024,
                },
              ]
            }
            src="featured.png"
            transition={null}
          >
            <div
              className="css-1d0r90g "
            />
          </Image>
        </ProductImage>
      </div>
    </Component>
  </ImageSlider>
</Connect(ImageSlider)>
`;

exports[`<ProductImageSlider /> should render image placeholder if no data is available 1`] = `
<Connect(ImageSlider)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ImageSlider
    images={null}
    navigate={[Function]}
    product={null}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <Component
      direction="DIRECTION_ALL"
      onPinchStart={[Function]}
      onTap={[Function]}
      options={
        Object {
          "recognizers": Object {
            "pinch": Object {
              "enable": true,
            },
          },
          "touchAction": "pan-x pan-y",
        }
      }
    >
      <div>
        <ProductImage
          alt={null}
          animating={true}
          forcePlaceholder={true}
          highestResolutionLoaded={[Function]}
          ratio={null}
          resolutions={
            Array [
              Object {
                "height": 440,
                "width": 440,
              },
              Object {
                "height": 1024,
                "width": 1024,
              },
            ]
          }
          src={null}
        >
          <div
            className="css-y5kkpq"
          >
            <div
              className="css-15j45j0"
              data-test-id="placeHolder"
            >
              <Placeholder
                className="css-19rc2oq"
              >
                <Icon
                  className="css-19rc2oq"
                  color={null}
                  content="<circle cx=\\"12\\" cy=\\"12\\" r=\\"3.2\\"/><path d=\\"M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z\\"/><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"/>"
                  size="inherit"
                  viewBox="0 0 24 24"
                >
                  <svg
                    className="css-yys9hb css-19rc2oq"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<circle cx=\\"12\\" cy=\\"12\\" r=\\"3.2\\"/><path d=\\"M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z\\"/><path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"/>",
                      }
                    }
                    style={
                      Object {
                        "fill": null,
                        "fontSize": "inherit",
                      }
                    }
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </Icon>
              </Placeholder>
            </div>
          </div>
        </ProductImage>
      </div>
    </Component>
  </ImageSlider>
</Connect(ImageSlider)>
`;

exports[`<ProductImageSlider /> should render image slider if all data is available 1`] = `
<Connect(ImageSlider)
  productId="913"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ImageSlider
    images={
      Array [
        "https://img-cdn.shopgate.com/image.jpg",
      ]
    }
    navigate={[Function]}
    product={
      Object {
        "ageRating": 0,
        "availability": Object {
          "state": "ok",
          "text": "In stock",
        },
        "baseProductId": null,
        "featuredImageUrl": "featured.png",
        "flags": Object {
          "hasChildren": false,
          "hasOptions": false,
          "hasVariants": false,
        },
        "highlight": true,
        "id": "913",
        "identifiers": Object {
          "sku": "857347363",
        },
        "liveshoppings": Array [],
        "manufacturer": "",
        "name": "This is a product with a very very long name -5-",
        "price": Object {
          "currency": "EUR",
          "info": "",
          "msrp": 0,
          "taxAmount": 0,
          "taxPercent": 0,
          "tiers": Array [],
          "unitPrice": 89,
          "unitPriceMax": 1000,
          "unitPriceMin": 0,
          "unitPriceNet": 89,
          "unitPriceStriked": 0,
          "unitPriceWithTax": 89,
        },
        "rating": Object {
          "average": 0,
          "count": 0,
          "reviewCount": 0,
        },
        "stock": Object {
          "ignoreQuantity": true,
          "info": "In stock",
          "maxOrderQuantity": 10000,
          "minOrderQuantity": 1,
          "orderable": true,
          "quantity": 0,
        },
      }
    }
    productId="913"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <Component
      direction="DIRECTION_ALL"
      onPinchStart={[Function]}
      onTap={[Function]}
      options={
        Object {
          "recognizers": Object {
            "pinch": Object {
              "enable": true,
            },
          },
          "touchAction": "pan-x pan-y",
        }
      }
    >
      <div>
        <ImageSlider
          autoPlay={false}
          className=""
          classNames={Object {}}
          controls={false}
          disabled={false}
          historyPath=""
          indicators={true}
          initialSlide={0}
          interval={3000}
          loop={true}
          maxIndicators={null}
          onSlideChange={[Function]}
          slidesPerView={1}
          snapItems={true}
        >
          <ProductImage
            alt={null}
            animating={false}
            forcePlaceholder={false}
            highestResolutionLoaded={[Function]}
            key=".$https=2//img-cdn.shopgate.com/image.jpg"
            ratio={null}
            resolutions={
              Array [
                Object {
                  "height": 440,
                  "width": 440,
                },
                Object {
                  "height": 1024,
                  "width": 1024,
                },
              ]
            }
            src="https://img-cdn.shopgate.com/image.jpg"
          >
            <Image
              alt={null}
              animating={false}
              backgroundColor="#fff"
              className=""
              forcePlaceholder={false}
              highestResolutionLoaded={[Function]}
              onError={[Function]}
              onLoad={null}
              ratio={null}
              resolutions={
                Array [
                  Object {
                    "height": 440,
                    "width": 440,
                  },
                  Object {
                    "height": 1024,
                    "width": 1024,
                  },
                ]
              }
              src="https://img-cdn.shopgate.com/image.jpg"
              transition={null}
            >
              <div
                className="css-1d0r90g "
              />
            </Image>
          </ProductImage>
        </ImageSlider>
      </div>
    </Component>
  </ImageSlider>
</Connect(ImageSlider)>
`;
